### PR TITLE
[velero] Upgrade CRDs API version to apiextensions.k8s.io/v1

### LINKED ISSUE #203: Any plan to update API version to v1?
Hi,
please.

**Describe the problem/challenge you have**
When running `helm` with latest minikube configuration:

```
╰─ minikube start
😄  minikube v1.16.0 on Darwin 11.1
✨  Automatically selected the hyperkit driver
👍  Starting control plane node minikube in cluster minikube
🔥  Creating hyperkit VM (CPUs=2, Memory=4000MB, Disk=20000MB) ...
🐳  Preparing Kubernetes v1.20.0 on Docker 20.10.0 ...
    ▪ Generating certificates and keys ...
    ▪ Booting up control plane ...
    ▪ Configuring RBAC rules ...
🔎  Verifying Kubernetes components...
🌟  Enabled addons: storage-provisioner, default-storageclass
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
```

I receive warnings and nothing happen:

```
W0118 22:09:45.018506   57397 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0118 22:09:45.031113   57397 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0118 22:09:45.042564   57397 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0118 22:09:45.053022   57397 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0118 22:09:45.074617   57397 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0118 22:09:45.087791   57397 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0118 22:09:45.097837   57397 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0118 22:09:45.139409   57397 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0118 22:09:45.154605   57397 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0118 22:09:45.164905   57397 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0118 22:09:45.176159   57397 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
```
**Describe the solution you'd like**
I'd like to have CRDs updated to v1.

**Environment:**

- helm version (use `helm version`): Version:"v3.4.0"
- helm chart version and app version:
  - helm3 (use `helm list -n <YOUR NAMESPACE>`): 
NAME  	NAMESPACE	REVISION	UPDATED                             	STATUS	CHART        	APP VERSION
velero	velero   	1       	2021-01-18 22:09:47.306611 +0100 CET	failed	velero-2.14.5	1.5.2
- Kubernetes version (use `kubectl version`): 1.20.0
- Kubernetes installer & version:
- Cloud provider or hardware configuration: minikube
- OS (e.g. from `/etc/os-release`): macOS

### PR DESCRIPTION
**Pull request by jenting**
#### Special notes for your reviewer:

- Update helm chart minimum Kubernetes version requirement as v1.16.
- Bump CRDs API version to `apiextensions.k8s.io/v1`.
- Bump Velero to v1.6.3 which supports CRDs API version `apiextensions.k8s.io/v1`.
- Bump default kubectl version to `1.16.15`.
- Update helm chart owner/reviewer/maintainer.

fixes #203

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the values.yaml or README.md
- [x] Title of the PR starts with chart name (e.g. `[velero]`)

### CODE REVIEW

File: .github/workflows/lint-test.yaml [hunk 1]
```diff
@@ -5,6 +5,17 @@ on: pull_request
...
+      matrix:
+        # Latest k8s versions. There's no series-based tag, nor is there a latest tag.
+        k8s:
+          - 1.16.15
+          - 1.17.17
+          - 1.18.15
+          - 1.19.7
+          - 1.20.2
+          - 1.21.1
+          #- 1.22.0
```
Comment (damoon) [inline .github/workflows/lint-test.yaml:18]: The commit message includes 1.22.
This line excludes 1.22.
Is this a typo or on purpose?

Comment (jenting) [reply .github/workflows/lint-test.yaml:18]: This PR is not ready yet, wait for Velero v1.6.3 release.

**Comment by jenting** [comment]
@zubron could you please help remove the GitHub CI action that must pass `lint-test` as Required? Because I don't have permission to configure it.
I think GitHub did not consolidate all the sub-matrixes results (1.16...1.22) into the `lint-test`, I did not find the related GitHub setting on this.

**Review by sseago — APPROVED** [review]

**Review by zubron — APPROVED** [review]

**Comment by zubron** [comment]
@jenting I've removed that check so this can be merged now. Do you want me to add each of the lint test actions as required? Or a subset of them? Due to the way the branch protection rules are configured, you have to explicitly reference each action that is triggered from the matrix, so we would need to be update each time we change the versions of Kubernetes used. I'll look to see if there there is another way to make them required without having to explicitly reference each entry in the matrix.

**Comment by jenting** [comment]
> @jenting I've removed that check so this can be merged now. Do you want me to add each of the lint test actions as required? Or a subset of them? Due to the way the branch protection rules are configured, you have to explicitly reference each action that is triggered from the matrix, so we would need to be update each time we change the versions of Kubernetes used. I'll look to see if there there is another way to make them required without having to explicitly reference each entry in the matrix.

I'd prefer to keep the current way until we find another way without having to set up each lint test as required, thank you :smiley:

**Pull request merged**